src/main/rx/eleres.c

```diff
@@ -0,0 +1,730 @@
+
+            rfTxBuffer[4] &= 0x0F;
+            rfTxBuffer[4] |= (hdop << 4);
+            rfTxBuffer[8] &= 0x1F;
+            rfTxBuffer[8] |= ((hdop<<1) & 0xE0);
+            break;
+        }
+        telem_state++;
+    case 2:
+        if (sensors(SENSOR_GPS)) {
+            uint16_t gpsspeed =  (gpsSol.groundSpeed*9L)/250L;
+            int16_t course = (gpsSol.groundCourse+360)%360;
+#ifdef NAV
+            int32_t alt = getEstimatedActualPosition(Z);
+#else
+            int32_t alt = baro.BaroAlt;
+#endif
+            uint16_t tim = 0;
+
+            rfTxBuffer[0] = 0x47;
+            rfTxBuffer[1] = (STATE(GPS_FIX)<<4) | (gpsSol.numSat & 0x0F);
+            if(gpsSol.numSat > 15) rfTxBuffer[1] |= 0x80;
+            rfTxBuffer[2] = ((course>>8) & 0x0F) | ((gpsspeed>>4) & 0xF0);
+            rfTxBuffer[3] = course & 0xFF;
+            rfTxBuffer[4] = gpsspeed & 0xFF;
+
+            rfTxBuffer[5] = (alt/100) >>8;
+            rfTxBuffer[6] = (alt/100) & 0xFF;
+
+            rfTxBuffer[7] = tim>>8;
+            rfTxBuffer[8] = tim&0xFF;
+            break;
+        }
+        telem_state++;
+    default:
+        rfTxBuffer[0] = 'D';
+        memcpy(rfTxBuffer+1,&debug[0],2);
+        memcpy(rfTxBuffer+3,&debug[1],2);
+        memcpy(rfTxBuffer+5,&debug[2],2);
+        memcpy(rfTxBuffer+7,&debug[3],2);
+        telem_state = 0;
+        break;
+    }
+    txFull = 1;
+}
+
+//because we shared SPI bus I can't read data in real IRQ, so need to probe this pin from main idle
+rx_spi_received_e eleresDataReceived(uint8_t *payload)
+{
+    UNUSED(payload);
+
+    statusRegisters[0] = 0;
+    statusRegisters[1] = 0;
+
+    if (rxSpiCheckIrq())
+    {
+        statusRegisters[0] = rfmSpiRead(0x03);
+        statusRegisters[1] = rfmSpiRead(0x04);
+        return RX_SPI_RECEIVED_DATA;
+    }
+
+    return RX_SPI_RECEIVED_NONE;
+}
+
+static void parseStatusRegister(const uint8_t *payload)
+{
+    UNUSED(payload);
+    static uint16_t rssifil;
+    const timeMs_t irq_time = millis();
+
+    if((rfMode & RECEIVE) && (statusRegisters[0] & RF22B_RX_PACKET_RECEIVED_INTERRUPT))
+        rfMode |= RECEIVED;
+    if((rfMode & TRANSMIT) && (statusRegisters[0] & RF22B_PACKET_SENT_INTERRUPT))
+        rfMode |= TRANSMITTED;
+    if((rfMode & RECEIVE) && (statusRegisters[1] & RF22B_VALID_SYNCWORD_INTERRUPT))
+        rfMode |= PREAMBLE;
+
+    if(rfMode & RECEIVED) {
+        if(bkgLocEnable < 2) {
+            lastPackTime = irq_time;
+            nextPackTime = irq_time + channelHoppingTime;
+        }
+        rxSpiReadCommandMulti(0x7f,0x00,rfRxBuffer,DATA_PACKAGE_SIZE);
+        rxReset();
+        rfMode = RECEIVE;
+        if ((rfRxBuffer[0] & 127) == 'S') {
+            firstRun = 0;
+            goodFrames++;
+
+            if ((rfRxBuffer[21] & 0xF0) == 0x20)
+                hoppingChannel = rfRxBuffer[21] & 0x0F;
+
+            channelHoppingTime = (rfRxBuffer[20] & 0x0F)+18;
+            dataReady |= 1;
```

Comment:
Can you replace the magic numbers associated with `dataReady` with `#define`s. Ie instead of
`dataReady |= 1;` use `dataReady |= SOME_MEANINGFUL_NAME;`. Likewise for 2, 4 etc.